src/main/kotlin/app/revanced/patches/youtube/misc/video/information/patch/VideoInformationPatch.kt

```diff
@@ -96,34 +104,55 @@ class VideoInformationPatch : BytecodePatch(
         }
 
         /*
-        Inject call for video id
+         * Inject call for video id
          */
         val videoIdMethodDescriptor = "$INTEGRATIONS_CLASS_DESCRIPTOR->setVideoId(Ljava/lang/String;)V"
         VideoIdPatch.injectCall(videoIdMethodDescriptor)
         VideoIdPatch.injectCallBackgroundPlay(videoIdMethodDescriptor)
 
         /*
-        Set the video time method
-        */
+         * Set the video time method
+         */
         with(PlayerControllerSetTimeReferenceFingerprint.result!!) {
             timeMethod = context.toMethodWalker(method)
                 .nextMethod(scanResult.patternScanResult!!.startIndex, true)
                 .getMethod() as MutableMethod
         }
 
         /*
-        Set the high precision video time method
+         * Set the high precision video time method
          */
         highPrecisionTimeMethod =
             (object : MethodFingerprint("V", null, listOf("J", "J", "J", "J", "I", "L"), null) {}).also {
                 it.resolve(context, VideoTimeFingerprint.result!!.classDef)
             }.result!!.mutableMethod
 
         /*
-        Hook the methods which set the time
+         * Hook the methods which set the time
          */
         highPrecisionTimeHook(INTEGRATIONS_CLASS_DESCRIPTOR, "setVideoTimeHighPrecision")
 
+
+        /*
+         * Hook the user playback speed selection
+         */
+        OnPlaybackSpeedItemClickFingerprint.result?.apply {
+            speedSelectionInsertMethod = mutableMethod
+            speedSelectionInsertIndex = scanResult.patternScanResult!!.startIndex - 3
+            speedSelectionValueRegister =
+                (mutableMethod.instruction(speedSelectionInsertIndex) as FiveRegisterInstruction).registerD
+
+            val speedSelectionMethodInstructions = mutableMethod.implementation!!.instructions
+            setPlaybackSpeedContainerClassFieldReference =
+                getReference(speedSelectionMethodInstructions, -1, Opcode.IF_EQZ)
+            setPlaybackSpeedClassFieldReference =
+                getReference(speedSelectionMethodInstructions, 1, Opcode.IGET)
+            setPlaybackSpeedMethodReference =
+                getReference(speedSelectionMethodInstructions, 2, Opcode.IGET)
+        } ?: return OnPlaybackSpeedItemClickFingerprint.toErrorResult()
+
+        userSelectedVideoSpeedHook(INTEGRATIONS_CLASS_DESCRIPTOR, "userSelectedPlaybackSpeed")
```

Comment:
Here, you refer to the playback speed as video speed, though everywhere else it is referred to as playback speed. It is likely better to stick to that naming for consistency reasons.